platform/source/ATCmdParser.cpp

```diff
@@ -259,17 +172,24 @@ bool ATCmdParser::vrecv(const char *response, std::va_list args)
         int j = 0;
 
         while (true) {
+            // Ran out of space
+            if (j + 1 >= _buffer_size - offset) {
+                return -1;
+            }
+
             // If just peeking for OOBs, and at start of line, check
             // readability
-            if (!response && j == 0 && !_fh->readable()) {
-                return false;
+            if (!issacnf && !response && j == 0 && !_fh->readable()) {
```

Comment:
I see that old `vrecv` didn't have this check, but actually I don't see why we would not return an error if the file handle (that is - serial port) is not readable, no matter if we're reading single or multiple lines?
Unless there is a good reason, I suggest to return error regardless of the flag.

Reply:
Done